graylog2-web-interface/src/views/types.d.ts

```diff
@@ -116,6 +120,7 @@ interface VisualizationType {
   displayName: string;
   component: VisualizationComponent;
   config?: VisualizationConfigDefinition;
+  validate?: (formValues: WidgetConfigFormValues) => { [key: string]: string | any };
```

Comment:
We could type `validate` like this:
`validate?: (formValues: WidgetConfigFormValues) => FormikErrors<WidgetConfigFormValues>`
to ensure we are always returning the errors in a format Formik can work with.